walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -138,4 +146,60 @@
 
         return ApiFutures.transform(snapshotFuture, jobFunction, MoreExecutors.directExecutor());
     }
+
+    /**
+     * Gets all the jobs given the params from the database.
+     * Currently, they can only be sorted/filtered by salary.
+     *
+     * @param jobQuery The job query object with all the filtering/sorting params.
+     * @return Future of the JobPage object.
+     */
+    public static Future<JobPage> fetchJobPage(JobQuery jobQuery) throws IOException {
+        CollectionReference jobsCollection = FireStoreUtils.getFireStore().collection(JOB_COLLECTION);
+
+        // TODO(issue/62): support other filters
+        if (!jobQuery.getSortBy().equals(Filter.SALARY)) {
+            throw new UnsupportedOperationException("currently this app only supports sorting/filtering by salary");
+        }
+
+        Query query = jobsCollection.whereEqualTo(JOB_STATUS_FIELD, JobStatus.ACTIVE.name())
+            .whereGreaterThanOrEqualTo(SALARY_FIELD, jobQuery.getMinLimit())
+            .whereLessThanOrEqualTo(SALARY_FIELD, jobQuery.getMaxLimit())
+            .orderBy(SALARY_FIELD, Order.getQueryDirection(jobQuery.getOrder()));
+
+        SingaporeRegion region = jobQuery.getRegion();
+        if (!region.equals(SingaporeRegion.ENTIRE)) {
+            query = query.whereEqualTo(REGION_FIELD, region.name());
+        }
+
+        // TODO(issue/34): add to the query to include pagination (using pageSize and pageIndex)
+
+        return ApiFutures.transform(
+            query.get(),
+            querySnapshot -> {
+                if (querySnapshot == null) {
+                    return new JobPage(ImmutableList.of(), 0, Range.between(0, 0));
```

Comment:
please annotate empty lists and naked ints. (same below) `/* paramName= */ImmutableList.of()`